addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/GetForPostScanRule.java

```diff
@@ -116,7 +120,20 @@ public void scan() {
             return;
         }
 
-        if (newRequest.getResponseBody().equals(baseMsg.getResponseBody())) {
+        ComparableResponse baseMsgComparableResponse =
+                new ComparableResponse(
+                        baseMsg.getResponseHeader().getStatusCode(),
+                        baseMsg.getResponseBody().toString(),
+                        Map.of(),
+                        "");
```

Comment:
Extract a method that has as parameter the HTTP message.

Reply:
I was actually suggesting this for the creation of the `ComparableResponse` since that's exactly the same for both messages.